monkey/monkey_island/cc/services/agent_binary_service/flask_resources/agent_binaries_masque.py

```diff
@@ -0,0 +1,51 @@
+import logging
+from http import HTTPStatus
+
+from flask import make_response, request
+from flask_security import auth_token_required, roles_accepted
+
+from common import OperatingSystem
+from monkey_island.cc.flask_utils import AbstractResource
+from monkey_island.cc.services.authentication_service import AccountRole
+
+from .. import IAgentBinaryService
+
+logger = logging.getLogger(__name__)
+
+
+class AgentBinariesMasque(AbstractResource):
+    urls = ["/api/agent-binaries/<string:os>/masque"]
+
+    def __init__(self, agent_binary_service: IAgentBinaryService):
+        self._agent_binary_service = agent_binary_service
+
+    @auth_token_required
+    @roles_accepted(AccountRole.ISLAND_INTERFACE.name)
+    def get(self, os):
+        try:
+            operating_system = OperatingSystem[os.upper()]
+
+            masque = self._agent_binary_service.get_masque(operating_system)
+            return make_response(
+                masque, HTTPStatus.OK, {"Content-Type": "application/octet-stream"}
+            )
+
+        except KeyError as err:
+            message = f'Could not get masque for OS "{os}": {err}'
+            logger.error(message)
+            return make_response({"error": message}, HTTPStatus.BAD_REQUEST)
```

Comment:
We can reduce the scope of this try/except to just the statement that raises a key error.